File: atividade-4/031_intersec/031_intersec.hs
module Main where

-- Faça uma função que receba duas listas a e b
-- e retorne uma lista das chaves que a e b possuem em comum

intersec :: [Int] -> [Int] -> [Int]
intersec xs ys = [y | y <- ys, (elem y xs)]

main = do
  a <- readLn :: IO [Int]
  b <- readLn :: IO [Int]
  print $ intersec a b

File: atividade-3/modulo-4/031_intersec/031_intersec.js
//Faça uma função que receba duas listas a e b
//e retorne uma lista das chaves que a e b possuem em comum

const list = [2, 5, 4, 9, 6, 10, 11, 12];
const list2 = [1, 2, 3, 4, 7, 42, 31, 20, 6];

const intersec = (xs, ys) => ys.filter((y) => xs.includes(y));

console.log(intersec(list, list2));
